src/cursor/mod.rs

```diff
@@ -170,7 +170,7 @@ impl GetMoreProviderResult for ImplicitSessionGetMoreResult {
 /// This is to be used with cursors associated with implicit sessions.
 enum ImplicitSessionGetMoreProvider {
```

Comment:
Clippy started complaining that the `Executing` variant was much bigger than the `Idle` one and suggested we `Box` the `Idle` one to fix this. I believe this is because, after the switch to `indexmap` in `bson`, the size of `Document` got a little bigger and was enough to trip the lint here.

Reply:
I'm actually getting this locally (without the BSON update), I think it's just a new complaint from clippy. Using `Box` sgtm.